diff --git a/cuda/bfs/Makefile b/cuda/bfs/Makefile
index 6ac5bde..2c38042 100644
--- a/cuda/bfs/Makefile
+++ b/cuda/bfs/Makefile
@@ -6,7 +6,7 @@ INCLUDE := $(CUDA_DIR)/include
 
 SRC = bfs.cu
 
-EXE = bfs.out
+EXE = bfs
 
 release: $(SRC)
 	$(CC) $(SRC) -o $(EXE) -I$(INCLUDE) -I../util -L$(CUDA_LIB_DIR) -DTIMING
diff --git a/cuda/cfd/Makefile b/cuda/cfd/Makefile
index 76df25d..f62470a 100755
--- a/cuda/cfd/Makefile
+++ b/cuda/cfd/Makefile
@@ -14,17 +14,17 @@ endif
 all: euler3d euler3d_double  pre_euler3d  pre_euler3d_double 
 
 euler3d: euler3d.cu
-	nvcc $(KERNEL_DIM) -O2 -Xptxas -v --gpu-architecture=compute_20 --gpu-code=compute_20 euler3d.cu -o euler3d -I$(CUDA_SDK_PATH)/common/inc  -L$(CUDA_SDK_PATH)/lib $(CUTIL_LIB)
+	nvcc $(KERNEL_DIM) -O2 -Xptxas -v --gpu-architecture=compute_30 --gpu-code=compute_30 euler3d.cu -o euler3d -I$(CUDA_SDK_PATH)/common/inc  -L$(CUDA_SDK_PATH)/lib $(CUTIL_LIB)
 
 euler3d_double: euler3d_double.cu
-	nvcc -Xptxas -v -O3 --gpu-architecture=compute_20 --gpu-code=compute_20 euler3d_double.cu -o euler3d_double -I$(CUDA_SDK_PATH)/common/inc  -L$(CUDA_SDK_PATH)/lib $(CUTIL_LIB)
+	nvcc -Xptxas -v -O3 --gpu-architecture=compute_30 --gpu-code=compute_30 euler3d_double.cu -o euler3d_double -I$(CUDA_SDK_PATH)/common/inc  -L$(CUDA_SDK_PATH)/lib $(CUTIL_LIB)
 
 
 pre_euler3d: pre_euler3d.cu
-	nvcc -Xptxas -v -O3 --gpu-architecture=compute_20 --gpu-code=compute_20 pre_euler3d.cu -o pre_euler3d -I$(CUDA_SDK_PATH)/common/inc  -L$(CUDA_SDK_PATH)/lib $(CUTIL_LIB)
+	nvcc -Xptxas -v -O3 --gpu-architecture=compute_30 --gpu-code=compute_30 pre_euler3d.cu -o pre_euler3d -I$(CUDA_SDK_PATH)/common/inc  -L$(CUDA_SDK_PATH)/lib $(CUTIL_LIB)
 
 pre_euler3d_double: pre_euler3d_double.cu
-	nvcc -Xptxas -v -O3 --gpu-architecture=compute_20 --gpu-code=compute_20 pre_euler3d_double.cu -o pre_euler3d_double -I$(CUDA_SDK_PATH)/common/inc  -L$(CUDA_SDK_PATH)/lib $(CUTIL_LIB)
+	nvcc -Xptxas -v -O3 --gpu-architecture=compute_30 --gpu-code=compute_30 pre_euler3d_double.cu -o pre_euler3d_double -I$(CUDA_SDK_PATH)/common/inc  -L$(CUDA_SDK_PATH)/lib $(CUTIL_LIB)
 
 
 clean:
diff --git a/cuda/dwt2d/Makefile b/cuda/dwt2d/Makefile
index 29c8b6c..daadbaa 100755
--- a/cuda/dwt2d/Makefile
+++ b/cuda/dwt2d/Makefile
@@ -9,7 +9,7 @@ cudagdb=0 #compile for use with cuda-gdb, note that 'dbg' must be 1 as well
 performancetest=1
 
 # NVCC Options
-NVCCFLAGS += -arch sm_20
+NVCCFLAGS += -arch sm_30
 
 # Files
 CFILES := 
@@ -109,4 +109,4 @@ $(EXECUTABLE): $(COBJS) $(CXXOBJS) $(CUOBJS)
 
 clean:
 	rm -f $(COBJS) $(CXXOBJS) $(CUOBJS) $(EXECUTABLE)
-	rm *.bmp.dwt.*		
\ No newline at end of file
+	rm *.bmp.dwt.*		
diff --git a/cuda/hotspot3D/Makefile b/cuda/hotspot3D/Makefile
index f08f693..6e96a5c 100755
--- a/cuda/hotspot3D/Makefile
+++ b/cuda/hotspot3D/Makefile
@@ -10,7 +10,7 @@ EXE = 3D
 
 OUTPUT = *.out
 
-FLAGS = -g -G #-arch sm_20 --ptxas-options=-v
+FLAGS = -g -G #-arch sm_30 --ptxas-options=-v
 release: $(SRC)
 	$(CC) $(KERNEL_DIM) $(FLAGS) $(SRC) -o $(EXE) -I$(INCLUDE) -L$(CUDA_LIB_DIR) 
 
diff --git a/cuda/hybridsort/Makefile b/cuda/hybridsort/Makefile
index 83e37e3..67d96ac 100644
--- a/cuda/hybridsort/Makefile
+++ b/cuda/hybridsort/Makefile
@@ -1,7 +1,7 @@
 include ../../common/make.config
 CC := $(CUDA_DIR)/bin/nvcc
 
-CC_FLAGS = -arch=sm_20
+CC_FLAGS = -arch=sm_30
 
 ifdef VERIFY
 override VERIFY = -DVERIFY
diff --git a/cuda/kmeans/Makefile b/cuda/kmeans/Makefile
index 7914bfa..330411c 100755
--- a/cuda/kmeans/Makefile
+++ b/cuda/kmeans/Makefile
@@ -24,7 +24,7 @@ endif
 kmeans: cluster.o getopt.o kmeans.o kmeans_clustering.o kmeans_cuda.o rmse.o
 	$(CC) $(CC_FLAGS) cluster.o getopt.o kmeans.o kmeans_clustering.o kmeans_cuda.o rmse.o -o kmeans -L$(CUDA_LIB_DIR) -lcuda -lcudart -lm
 
-%.o: %.[ch]
+%.o: %.c
 	$(CC) $(CC_FLAGS) $< -c
 
 kmeans_cuda.o: kmeans_cuda.cu 
diff --git a/cuda/lavaMD/makefile b/cuda/lavaMD/makefile
index 3944253..1c9188c 100755
--- a/cuda/lavaMD/makefile
+++ b/cuda/lavaMD/makefile
@@ -18,7 +18,7 @@ OMP_FLAG = -fopenmp
 
 CUD_C = $(CUDA_DIR)/bin/nvcc
 # OMP_FLAG = 	-Xcompiler paste_one_here
-CUDA_FLAG = -arch sm_13
+CUDA_FLAG = -arch sm_30
 
 # link objects (binaries) together
 a.out:		main.o \
diff --git a/cuda/leukocyte/CUDA/Makefile b/cuda/leukocyte/CUDA/Makefile
index 3086f62..7e8f8a8 100755
--- a/cuda/leukocyte/CUDA/Makefile
+++ b/cuda/leukocyte/CUDA/Makefile
@@ -31,7 +31,7 @@ endif
 leukocyte: detect_main.o avilib.o find_ellipse.o find_ellipse_kernel.o track_ellipse.o track_ellipse_kernel.o misc_math.o $(MATRIX_DIR)/meschach.a
 	$(CC) $(CC_FLAGS) -lm avilib.o find_ellipse.o find_ellipse_kernel.o track_ellipse.o track_ellipse_kernel.o misc_math.o detect_main.o -o leukocyte $(MATRIX_DIR)/meschach.a -L$(CUDA_LIB_DIR) -lm -lcuda -lcudart
 
-%.o: %.[ch]
+%.o: %.c
 	$(CC)  $(OUTPUT) $(CC_FLAGS) $< -c
 
 detect_main.o: detect_main.c find_ellipse.h track_ellipse.h avilib.h 
diff --git a/cuda/lud/cuda/Makefile b/cuda/lud/cuda/Makefile
index 492edd2..47e5e35 100644
--- a/cuda/lud/cuda/Makefile
+++ b/cuda/lud/cuda/Makefile
@@ -8,7 +8,7 @@ DEFS += \
 NVCCFLAGS += -I../common \
 			 -O3 \
 			 -use_fast_math \
-			 -arch=sm_20 \
+			 -arch=sm_30 \
 			 -lm \
 			 $(SPACE)
 
@@ -38,7 +38,7 @@ all : $(EXECUTABLE)
 
 clang: $(SRC)
 	clang++ lud.cu lud_kernel.cu ../common/common.c -o $(EXECUTABLE) \
-		-I../common -I../../util --cuda-gpu-arch=sm_20 \
+		-I../common -I../../util --cuda-gpu-arch=sm_30 \
 		-L/usr/local/cuda/lib64 -lcudart_static -ldl -lrt -pthread -DTIMING
 
 $(EXECUTABLE) : $(OBJS)
diff --git a/cuda/mummergpu/src/suffix-tree.cpp b/cuda/mummergpu/src/suffix-tree.cpp
index 7f8d141..3c9e25c 100755
--- a/cuda/mummergpu/src/suffix-tree.cpp
+++ b/cuda/mummergpu/src/suffix-tree.cpp
@@ -15,6 +15,7 @@
 #include <errno.h>
 #include <assert.h>
 #include <stdint.h> 
+#include <unistd.h>
 
 #define ulong4 uint32_t
 #define uint4 uint32_t
diff --git a/cuda/nn/Makefile b/cuda/nn/Makefile
index d218050..2784968 100644
--- a/cuda/nn/Makefile
+++ b/cuda/nn/Makefile
@@ -13,7 +13,7 @@ nn : nn_cuda.cu
 	$(CC) -o nn nn_cuda.cu
 
 clang: $(SRC)
-	clang++ nn_cuda.cu -o nn -I../util --cuda-gpu-arch=sm_20 \
+	clang++ nn_cuda.cu -o nn -I../util --cuda-gpu-arch=sm_30 \
 		-L/usr/local/cuda/lib64 -lcudart_static -ldl -lrt -pthread -DTIMING
 
 hurricane_gen : hurricane_gen.c
diff --git a/cuda/nw/Makefile b/cuda/nw/Makefile
index 29180ae..1ecafe7 100755
--- a/cuda/nw/Makefile
+++ b/cuda/nw/Makefile
@@ -12,7 +12,7 @@ release: $(SRC)
 	$(CC) ${KERNEL_DIM} $(SRC) -o $(EXE) -I$(INCLUDE) -L$(CUDA_LIB_DIR) 
 
 clang: $(SRC)
-	clang++ $(SRC) -o $(EXE) -I../util --cuda-gpu-arch=sm_20 \
+	clang++ $(SRC) -o $(EXE) -I../util --cuda-gpu-arch=sm_30 \
 		-L/usr/local/cuda/lib64 -lcudart_static -ldl -lrt -pthread -DTIMING
 
 enum: $(SRC)
diff --git a/cuda/particlefilter/Makefile b/cuda/particlefilter/Makefile
index a7db5ca..ec18015 100755
--- a/cuda/particlefilter/Makefile
+++ b/cuda/particlefilter/Makefile
@@ -9,10 +9,10 @@ INCLUDE := $(CUDA_DIR)/include
 all: naive float
 
 naive: ex_particle_CUDA_naive_seq.cu
-	$(CC) -I$(INCLUDE) -L$(CUDA_LIB_DIR) -lcuda -g -lm -O3 -use_fast_math -arch sm_13 ex_particle_CUDA_naive_seq.cu -o particlefilter_naive
+	$(CC) -I$(INCLUDE) -L$(CUDA_LIB_DIR) -lcuda -g -lm -O3 -use_fast_math -arch sm_30 ex_particle_CUDA_naive_seq.cu -o particlefilter_naive
 	
 float: ex_particle_CUDA_float_seq.cu
-	$(CC) -I$(INCLUDE) -L$(CUDA_LIB_DIR) -lcuda -g -lm -O3 -use_fast_math -arch sm_13 ex_particle_CUDA_float_seq.cu -o particlefilter_float
+	$(CC) -I$(INCLUDE) -L$(CUDA_LIB_DIR) -lcuda -g -lm -O3 -use_fast_math -arch sm_30 ex_particle_CUDA_float_seq.cu -o particlefilter_float
 
 clean:
 	rm particlefilter_naive particlefilter_float
diff --git a/cuda/pathfinder/Makefile b/cuda/pathfinder/Makefile
index 058a06e..369407d 100644
--- a/cuda/pathfinder/Makefile
+++ b/cuda/pathfinder/Makefile
@@ -11,7 +11,7 @@ release:
 	$(CC) $(SRC) -o $(EXE) -I$(INCLUDE) -L$(CUDA_LIB_DIR) 
 
 clang: $(SRC)
-	clang++ $(SRC) -o $(EXE) -I../util --cuda-gpu-arch=sm_20 \
+	clang++ $(SRC) -o $(EXE) -I../util --cuda-gpu-arch=sm_30 \
 		-L/usr/local/cuda/lib64 -lcudart_static -ldl -lrt -pthread -DTIMING
 
 enum:
diff --git a/cuda/streamcluster/Makefile b/cuda/streamcluster/Makefile
index 7d5624f..822959e 100755
--- a/cuda/streamcluster/Makefile
+++ b/cuda/streamcluster/Makefile
@@ -21,7 +21,7 @@ endif
 
 # make dp=1 compiles the CUDA kernels with double-precision support
 ifeq ($(dp),1)
-	NVCC_FLAGS += --gpu-name sm_13
+	NVCC_FLAGS += --gpu-name sm_30
 endif
 
 
